core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -260,6 +312,165 @@ public KeyStore getKeyStore(String keyStoreName) throws Exception {
+        primaryKeyStoreMetadata.setKeyStoreName(name);
+        primaryKeyStoreMetadata.setKeyStoreType(type);
+        primaryKeyStoreMetadata.setProvider(" ");
+        primaryKeyStoreMetadata.setPrivateStore(true);
+        return primaryKeyStoreMetadata;
+    }
+
+    /**
+     * This method is used to generate the file name of the public cert of a tenant.
+     *
+     * @param ksLocation Keystore location in the registry.
+     * @param uuid       UUID appender.
+     * @return file name of the public cert.
+     */
+    private String generatePubCertFileName(String ksLocation, String uuid) {
+
+        String tenantName = ksLocation.substring(ksLocation.lastIndexOf("/"));
+        for (KeystoreUtils.StoreFileType fileType : KeystoreUtils.StoreFileType.values()) {
+            String fileExtension = KeystoreUtils.StoreFileType.getExtension(fileType);
+            if (tenantName.endsWith(fileExtension)) {
+                tenantName = tenantName.replace(fileExtension, "");
+            }
+        }
+        return tenantName + "-" + uuid + ".cert";
+    }
+
+    /**
+     * Get keystore type.
+     *
+     * @param keyStoreName Keystore name.
+     * @return Keystore type.
+     * @throws SecurityException If an error occurs while retrieving the keystore type.
+     */
+    public String getKeyStoreType(String keyStoreName) throws SecurityException {
+
+        String keyStoreType;
+        ServerConfiguration serverConfig = ServerConfiguration.getInstance();
+        if (KeyStoreUtil.isPrimaryStore(keyStoreName)) {
+            keyStoreType =
+                    serverConfig.getFirstProperty(RegistryResources.SecurityManagement.SERVER_PRIMARY_KEYSTORE_TYPE);
+        } else if (KeyStoreUtil.isTrustStore(keyStoreName)) {
+            keyStoreType = serverConfig.getFirstProperty(RegistryResources.SecurityManagement.SERVER_TRUSTSTORE_TYPE);
+        } else {
+            String path = RegistryResources.SecurityManagement.KEY_STORES + "/" + keyStoreName;
+            try {
+                if (!registry.resourceExists(path)) {
+                    throw new SecurityException("Keystore " + keyStoreName + " not found at " + path);
+                }
+                Resource resource = registry.get(path);
+                keyStoreType = resource.getProperty(RegistryResources.SecurityManagement.PROP_TYPE);
+            } catch (RegistryException e) {
+                String msg = "Error when getting keyStore type for the keystore : " + keyStoreName;
+                log.error(msg, e);
+                throw new SecurityException(msg, e);
+            }
+        }
+        return keyStoreType;
+    }
+
+    /**
+     * Check the existence of the key store for given keystore name.
+     *
+     * @return True if the key store exists, false otherwise.
+     * @throws Exception If an error occurs while checking the existence of the key store.
+     */
+    public boolean isKeyStoreExists(String keyStoreName) throws SecurityException {
```

Comment:
Check whether we really need this method.. We may already do the isExists check in another layer, so we can avoid having this public method.